File: TDDMicroExercises.ProposedSolution/Java/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm
{
    public static final double LOW_PRESSURE_THRESHOLD = 17;
    public static final double HIGH_PRESSURE_THRESHOLD = 21;

    ISensor sensor;

    boolean alarmOn = false;
    
    public Alarm(){
    	this(new Sensor());
    }
    
    public Alarm(ISensor sensor) {
		this.sensor = sensor;
	}

	public void check()
    {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LOW_PRESSURE_THRESHOLD || HIGH_PRESSURE_THRESHOLD < psiPressureValue)
        {
            alarmOn = true;
        }
    }

    public boolean isAlarmOn()
    {
        return alarmOn; 
    }
}

File: Rust/tire_pressure_monitoring_system/src/alarm.rs
use crate::sensor::Sensor;

pub struct Alarm {
    low_pressure_threshold: f64,
    high_pressure_threshold: f64,
    sensor: Sensor,
    alarm_on: bool,
}

impl Alarm {
    pub fn new() -> Alarm {
        Alarm {
            low_pressure_threshold: 17.0,
            high_pressure_threshold: 21.0,
            sensor: Sensor::new(),
            alarm_on: false,
        }
    }

    pub fn check(&mut self) {
        let psi_pressure_value = self.sensor.pop_next_pressure_psi_value();

        if psi_pressure_value < self.low_pressure_threshold
            || psi_pressure_value > self.high_pressure_threshold
        {
            self.alarm_on = true;
        }
    }

    pub fn is_alarm_on(&self) -> bool {
        self.alarm_on
    }
}
